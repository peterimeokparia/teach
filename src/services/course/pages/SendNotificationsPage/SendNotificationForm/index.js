import { useState } from 'react';

const SendNotificationForm = ({
error,    
handleSendingPushMessage,
setHandleFormReset }) => {
const [ notificationMessage, setNotificationMessage ] = useState('');
const [ notificationTitle, setNotificationTitle ] = useState('');

const handleSubmit = (e) => {
  e.preventDefault();
  resetForm();
};

const sendPushNotificationMessage = ( title, message ) => {
  handleSendingPushMessage( title, message );
  resetForm();
};

const resetForm = () => {
  setNotificationTitle('');
  setNotificationMessage('');
  setHandleFormReset(true);
};

return (
        <div>
            <form onSubmit={e => handleSubmit(e)}>
                <label>
                  Push Notification Message Title
                  <input
                    name="notification-title"
                    type="text"
                    value={notificationTitle}
                    onChange={ e => setNotificationTitle( e.target.value ) }
                    placeholder={"push notification title"}
                  >  
                  </input>
                </label>
                <label>
                    Push Notification Message
                  <input
                      name="notification-message"
                      type="text"
                      value={notificationMessage}
                      onChange={ e => setNotificationMessage( e.target.value ) }
                      placeholder={"push notification message"}
                  >  
                  </input>
                  </label>
                  { error  && (<div className="error"> { error.message }</div>)}
                  <div></div>
                  <button 
                    type="submit" 
                    onClick={e => sendPushNotificationMessage( notificationTitle, notificationMessage )}
                  >
                  Send Message
                  </button>
            </form>
        </div>
); };

export default SendNotificationForm;