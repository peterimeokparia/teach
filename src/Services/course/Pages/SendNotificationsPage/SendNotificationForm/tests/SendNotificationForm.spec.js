import React from 'react';
import SendNotificationForm  from 'services/course/pages/SendNotificationsPage/SendNotificationForm';
import renderer from 'react-test-renderer';

describe('SendNotificationForm', () =>  {  

  beforeEach(() => {
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
        .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {
    const tree = renderer.create(
        <SendNotificationForm /> 
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});




