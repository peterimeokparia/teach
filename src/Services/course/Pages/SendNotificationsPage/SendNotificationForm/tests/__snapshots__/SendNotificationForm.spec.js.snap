// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendNotificationForm renders as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Push Notification Message Title
      <input
        name="notification-title"
        onChange={[Function]}
        placeholder="push notification title"
        type="text"
        value=""
      />
    </label>
    <label>
      Push Notification Message
      <input
        name="notification-message"
        onChange={[Function]}
        placeholder="push notification message"
        type="text"
        value=""
      />
    </label>
    <div />
    <button
      onClick={[Function]}
      type="submit"
    >
      Send Message
    </button>
  </form>
</div>
`;
